controllers/restic.go

```diff
@@ -53,16 +56,23 @@ func (r *VeleroReconciler) ReconcileResticDaemonset(log logr.Logger) (bool, erro
 	ds := &appsv1.DaemonSet{
 		ObjectMeta: getResticObjectMeta(r),
 	}
-
 	if velero.Spec.EnableRestic != nil && !*velero.Spec.EnableRestic {
-
-		if err := r.Get(r.Context, r.NamespacedName, ds); err == nil {
+		deleteContext := context.Background()
+		if err := r.Get(deleteContext, types.NamespacedName{
+			Name:      ds.Name,
+			Namespace: r.NamespacedName.Namespace,
+		}, ds); err == nil {
```

Comment:
Change this so `if err != nil` record event, and return.

Then run deletion logic outside of conditional.